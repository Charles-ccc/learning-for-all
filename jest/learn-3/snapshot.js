export const generateConfig = () => {
  return {
    server: 'http://192.168.0',
    port: '8080'
  }
}

export const generateConfig2 = () => {
  return {
    server: 'http://192.168.0',
    port: '8080',
    time: new Date()
  }
}