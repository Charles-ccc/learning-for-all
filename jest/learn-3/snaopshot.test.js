import { generateConfig, generateConfig2 } from "./snapshot";

// test('测试 generateConfig', () => {
//   expect(generateConfig()).toMatchSnapshot()
// })

// test('测试 generateConfig2', () => {
//   expect(generateConfig2()).toMatchSnapshot({
//     time: expect.any(Date)
//   })
// })

test("测试 generateConfig2", () => {
  // toMatchInlineSnapshot 将快照放在测试用例里面
  expect(generateConfig2()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "port": "8080",
      "server": "http://192.168.0",
      "time": Any<Date>,
    }
  `
  );
});
