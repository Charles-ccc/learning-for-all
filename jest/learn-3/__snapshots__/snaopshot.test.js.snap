// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateConfig 1`] = `
Object {
  "port": "8080",
  "server": "http://192.168.0",
}
`;

exports[`测试 generateConfig2 1`] = `
Object {
  "port": "8080",
  "server": "http://192.168.0",
  "time": Any<Date>,
}
`;
